### PR TITLE
Implement Plot Directive for ConfusionMatrix, PrecisionRecall and Decisionviz

### PR DESCRIPTION
**Pull request by lwgray**
Addresses Issue #687 

This PR implements Plot Directives for the ConfusionMatrix, PrecisionRecallCurve, and DecisionViz docs.  

NOTES 1
--------
The only issue that I ran into was that the old DecisionViz script produced two images but only one was included in the docs.  This is the same type of  issue that I've discussed with @rebeccabilbro @bbengfort  and I decided to include the second image in the plot directive (see description #737).

NOTES 2
---------
The confusion matrix when plotting with class names results in cut-off labels.  This was an open issue #613 but was closed because of forthcoming mpl updates.  However, I think this issue should be reopened.

### CODE REVIEW

File: docs/api/classifier/prcurve.rst [hunk 1]
```diff
@@ -64,24 +65,36 @@ To support multi-label classification, the estimator is wrapped in a `OneVsRestC
...
+    X = OrdinalEncoder().fit_transform(X)
+    y = LabelEncoder().fit_transform(y)
+
+    X_train, X_test, y_train, y_test = tts(X, y, test_size=0.2, shuffle=True)
+
+    # Create the visualizer, fit, score, and poof it
+
+    viz = PrecisionRecallCurve(
         MultinomialNB(), per_class=True, iso_f1_curves=True,
         fill_area=False, micro=False
```
Comment (bbengfort) [inline docs/api/classifier/prcurve.rst]: @lwgray now that #688 has been merged in, would you mind updating the code as follows:

```python
encoder = LabelEncoder()
y = encoder.fit_transform(y)

oz = PrecisionRecallCurve(
    MultinomialNB(), per_class=True, iso_f1_curves=True, fill_area=False, 
    micro=False, classes=encoder.classes_
)
```

This will be a good test to ensure the bug is fixed and is potentially a more realistic example! Thank you and @zjpoh for this enhancement!

Comment (lwgray) [reply docs/api/classifier/prcurve.rst]: done!
prcurve-3

Comment (bbengfort) [reply docs/api/classifier/prcurve.rst]: thank you!

**Comment by lwgray** [comment]
@rebeccabilbro when building docs...I noticed a few sklearn FutureWarnings regarding changing default values (ie n_estimators for RandomForest; multiclass and solver for LogisticRegression)...Do you want me to add to the code to address these Warnings... For example by specifically setting ie solver='liblinear', n_estimators=10, etc

**Comment by lwgray** [comment]
In addition to your PEP-8 suggestions, I went through and made the changes to remove the FutureWarnings.  Thanks for the review! 😸

**Comment by rebeccabilbro** [comment]
Thanks @lwgray; merged! 🎉
